### PR TITLE
Setup enhancements, including reading long_description from README.rst (#254)

### PR DESCRIPTION
**Pull request by pekkaklarck**
This pull requests consists of two commits that enhance setup.py:
- f110449 implements reading long_description value from README.rst (#254).
- 9c3b565 contains general clean-up but doesn't change functionality.

Feel free to cherry-pick only the former commit if you don't agree changes in the latter or are afraid they break something.

### CODE REVIEW

**Comment by pekkaklarck** [comment]
Travis build is failing but it seems to have been doing so already earlier.

File: setup.py [hunk 1]
```diff
@@ -1,46 +1,41 @@
 #!/usr/bin/env python
 
-import sys
-from os.path import join, dirname
+from os.path import abspath, dirname, join
 
-sys.path.append(join(dirname(__file__), 'src'))
```
Comment (emanlove) [inline setup.py:6]: Removal of this line breaks the install process as...

File: setup.py [hunk 2]
```diff
@@ -1,46 +1,41 @@
...
 from ez_setup import use_setuptools
```
Comment (emanlove) [inline setup.py:5]: ... this line requires ez_setup which we include under the src directory. We should actually remove the dependency on ez_setup.py and remove it from source.

**Comment by aaltat** [comment]
@pekkaklarck is this PR still valid one, or should we close it?

**Comment by pekkaklarck** [comment]
Enhancements themselves are valid but but due to conflicts it's easier to just re-do them than trying to fix this PR.

**Pull request closed without merging**